app/jobs/csv_export_job.rb

```diff
@@ -34,22 +34,20 @@ def deploy_csv_export(csv_export)
     filename = csv_export.path_file
     filter = csv_export.filters
 
-    @deploys = Deploy.joins(:stage, :job).where(filter)
+    @deploys = Deploy.joins(
+      'INNER JOIN stages ON stages.id = deploys.stage_id INNER JOIN jobs ON jobs.id = deploys.job_id'
+    ).unscope(where: :deleted_at).where(filter)
```

Comment:
With the different association, optimized the query to prefetch associations during the join except for users and buddy.  Otherwise the in-memory table probably gets a bit large.